cron/warnStaleWorkspaces.js

```diff
@@ -0,0 +1,29 @@
+const util = require('util');
+
+const config = require('../lib/config');
+const logger = require('../lib/logger');
+const workspaceHelper = require('../lib/workspace');
+const sqldb = require('@prairielearn/prairielib/sql-db');
+const sqlLoader = require('@prairielearn/prairielib/sql-loader');
+
+const sql = sqlLoader.loadSqlEquiv(__filename);
+
+module.exports = {};
+
+module.exports.run = function(callback) {
+    util.callbackify(async () => {
+        const params = {
+            launched_timeout_sec: config.workspaceLaunchedTimeoutSec,
+            launched_timeout_warn_sec: config.workspaceLaunchedTimeoutWarnSec,
+        };
+        const result = await sqldb.queryAsync(sql.select_almost_stale_workspaces, params);
+        const workspaces = result.rows;
+        for (const workspace of workspaces) {
+            const launched_at = new Date(workspace.launched_at);
+            const timeout_at = new Date(launched_at.getTime() + config.workspaceLaunchedTimeoutSec * 1000);
+            const time_to_timeout_min = (timeout_at - Date.now()) / 1000 / 60;
```

Comment:
Could we shift the calculation of `time_to_timeout_min` into the SQL? I think that the JS is completely correct here, but I generally prefer to keep time/date calculations in the SQL to avoid surprises with JS date handling.

Reply:
done in 89c13c3830c4998c6b926ed16f18c3b31dfdb1f0